src/mainwindow.cpp

```diff
@@ -494,21 +494,53 @@ void MainWindow::applyCompanion(const Extensions::CompanionData &data)
 
     if (isUntitled() && !isTextChanged())
     {
-        QString meta = data.toMetaString();
-        meta.prepend("\n");
-        meta.append("Powered by CP Editor (https://github.com/cpeditor/cpeditor)");
+        QString comments = SettingsHelper::getCompetitiveCompanionHeadComments().replace(
+            "${time}", QDateTime::currentDateTime().toString("yyyy-MM-dd HH:mm:ss"));
+
+        auto it = QRegularExpression(R"(\$\{json\..+?\})").globalMatch(comments);
+
+        QString finalComments = "\n";
+        int lastEnd = 0;
+
+        while (it.hasNext())
+        {
+            auto match = it.next();
+            finalComments += comments.mid(lastEnd, match.capturedStart() - lastEnd);
+            auto path = match.captured().mid(7, match.capturedLength() - 8).split(".");
+            auto value = QJsonValue::fromVariant(data.doc.toVariant());
+            for (auto attr : path)
+                value = value[attr];
+            if (value.isUndefined())
+            {
+                log->error(tr("Companion"),
```

Comment:
Competitive Companion is a name of a software product. Does it needs to be translated?